src/logging/Logger.cpp

```diff
@@ -1,53 +1,158 @@
-#include "Logger.hpp"
 #include <boost/log/attributes/constant.hpp>
-#include <boost/log/attributes/mutable_constant.hpp>
+#include <boost/log/attributes/function.hpp>
 #include <boost/log/attributes/named_scope.hpp>
-#include <boost/log/core.hpp>
-#include <boost/log/expressions.hpp>
+#include <boost/log/attributes/timer.hpp>
+#include <boost/log/sources/severity_feature.hpp>
 #include <boost/log/trivial.hpp>
 #include <boost/log/utility/setup/common_attributes.hpp>
-#include <iosfwd>
 #include <utility>
+#include <utils/assertion.hpp>
+
+#include "logging/LogConfiguration.hpp"
+#include "logging/Logger.hpp"
 
 namespace precice {
 namespace logging {
 
+/// A feature that adds \ref LogLocation info to log attributes
+template <class BaseT>
+class precice_feature : public BaseT {
+public:
+  using char_type       = typename BaseT::char_type;
+  using threading_model = typename BaseT::threading_model;
+
+  precice_feature()                             = default;
+  precice_feature(const precice_feature &other) = default;
+  template <typename ArgsT>
+  precice_feature(ArgsT const &args){};
+
+  using open_record_lock = typename boost::log::strictest_lock<
+      boost::lock_guard<threading_model>,
+      typename BaseT::open_record_lock,
+      typename BaseT::add_attribute_lock,
+      typename BaseT::remove_attribute_lock>::type;
+
+protected:
```

Comment:
Considering that we inherit things, we should add a virtual destructor.

Reply:
This is compile-time inheritance using RCTP, which does not require a virtual dtor. It would also only be necessary in the base class.